### DISCUSSION TITLE
compiling on windows

### DISCUSSION THREAD

**Discussion by bsmokeman** [Brewken/brewken#30] [General]
I noticed from some conversations on brewtarget github discussions, you have some experience compiling on windows.  Which direction did you go for editing, and compiling.  I'm using the open source QT installation, with qt creator to look at code, and figure things out, but my compilation attempts have failed so far.  I've followed the how to compile for windows, but I keep getting various errors depending on how I try to compile.  I've tried the qt creator project compile, cmake gui, msys2.  I'm sure I have all the required libraries installed, so it's gotta be a simple configuration.  Will it compile with x86_64, or will I have to install the win32 libs?  Is there a way to get qt creator configured to compile the project, or is cmake gui, or msys2 the only way to get it to compile?  Since QT creator isn't configured correctly, I have a lot of digging to do.
Just curious how you are doing it, I'm still poking around.
I could move to a linux vm, and work off of that environment, however we plan to run from windows based clients, but the main server could be linux, and I actually prefer that.  Dolibarr will be a linux server install as well, and was hoping to install both on the same linux vm, making the sync integration much easier.
thanks, for any info you have that can help...
Barry Smoke

**Comment by matty0ung** [comment]
I tried to write down what worked for me at https://github.com/Brewken/brewken/wiki/Building-on-Windows, ie using MSYS2, though, as noted on that page, that might not be how you want to do things if Windows is your primary platform.  The build should work in 64-bits up to and including the point where you can install and run the software locally.  (Building 64-bit Windows packages for distribution would still need some work because of some issue with NSIS, but I haven't looked at that in detail.)

I wouldn't expect Qt creator to be able to build things out of the box because (if I understood correctly) it's likely geared up for qmake rather than CMake.  You definitely want to be using CMake.  I don't know about the CMake GUI, as I've not used it.

See also https://github.com/Brewken/brewken/wiki/Development:-Building for some notes on the make commands being a bit different on Windows.  (You just have to call cmake directly rather than calling make on a Makefile generated by CMake.  We could probably fix this, but it didn't make it to the top of the priority list yet!

You might find that switching compilers, eg to MSVC, would throw up one or two build errors and/or warnings (as this is what I found getting the Mac build to work with Clang compiler) but those should all be fixable with small code changes.

Happy to offer any suggestions if you post errors here.
